docs/deployment/universal-profile.md

```diff
@@ -159,31 +200,242 @@ await lspFactory.LSP3UniversalProfile.deploy({
+    },
+    KeyManager: {
+        libAddress: '0x6c1F3Ed2F99054C88897e2f32187ef15c62dC560'
+    },
+})
+```
+
+### Proxy Deployment
+
+Allows you to determine whether your contract should be deployed as a **minimal proxy contract** based on [EIP1167](https://eips.ethereum.org/EIPS/eip-1167) or a full contract with a constructor.
+
+```javascript
+lspFactory.LSP3UniversalProfile.deploy({...}, {
+    ERC725Account: {
+        deployProxy: false,
+    },
+})
+```
+
+A proxy contract is a lightweight contract which inherits its logic by referencing the address of a contract already deployed on the blockchain. This allows cheaper deployment of Universal Profiles because only the proxy contract needs to be deployed.
+
+LSPFactory stores base contract addresses inside the [version file](https://github.com/lukso-network/tools-lsp-factory/blob/main/src/versions.json). If no version is specified in the `version` parameter, the latest available base contract version will be used.
+
+:::info Info
+LSPFactory stores base contract addresses for different versions [internally](https://github.com/lukso-network/tools-lsp-factory/blob/main/src/versions.json) and uses the latest available version if no version is specified.
+:::
+
+If you do not want your contract to use proxy deployment you can set `deployProxy` to `false`. This will deploy a 'full' contract with a constructor rather than using a proxy deployment with initializer. When set to `false`, the latest available contract version will be deployed.
+
+`deployProxy` defaults to `true` for `ERC725Account` and `LSP6KeyManager` and `false` for `UniversalReceiverDelegate`.
+
+When using proxy deployment, LSPFactory will check that there is bytecode deployed at the base contract address before deploying. If there is none, a new base contract will be deployed and referenced in the proxy contract. This is useful when using LSPFactory on a local development network like Hardhat where there will be no pre-deployed base contracts.
+
+#### Universal Receiver Delegate proxy deployment
+
+The `UniversalReceiverDelegate` is a logic contract which writes to the Universal Profile when it recieves some asset. This is not specific to any particular Universal Profile so the same `UniversalReceiverDelegate` contract can be used for multiple different Universal Profile deployments.
+
+By default LSPFactory will use the latest available version of `UniversalReceiverDelegate` version stored in the [version file](https://github.com/lukso-network/tools-lsp-factory/blob/main/src/versions.json). This address is used directly on the Universal Profile and is given the `SETDATA` LSP6 permission. This means that no `UniversalReceiverDelegate` contract needs to be deployed when deploying a Universal Profile which further saves on the gas cost of Universal Profile deployment.
+
+To specify that your Universal Profile should use proxy deployment when deplying the `UniversalReceiverDelegate` contract, set `deployProxy` to `true`.
```

Comment:
typo:

~`deplying`~ -> **`deploying`**

We have to discuss if this is necessary.
Since there is already one deployed Universal Receiver Delegate contract for all UP. So putting a proxy in front might not be necessary.